### PR TITLE
Update discord.js

### PR DESCRIPTION
**Pull request by ricardodevries**
The package discord.js@12.5.3 is no longer supported. This pull request upgrades the package to v13.8.0 and changes the code to reflect the upgrade.

It might be that the client needs more intents but so far we got _GUILD_MESSAGES_ and _GUILD_MESSAGE_REACTIONS_ (Based on the available events in `src/events/discord`).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "axios": "^0.21.4",
     "body-parser": "^1.19.0",
     "cors": "^2.8.5",
-    "discord.js": "^12.4.0",
+    "discord.js": "^13.8.0",
```
Comment (ricardodevries) [inline package.json:34]: I don't see what Node.js version is used on Heroku with p4nth3rb0t-mainframe. Node.js 16.6.0 or newer is required for discord.js@13. To use [Node.js 16 with Heroku](https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version) you need to add the following to `package.json`:

```json
"engines": {
  "node": "16.x"
}
```

File: src/events/discord/message.ts [hunk 2]
```diff
@@ -86,12 +86,9 @@ discord.on("message", async (message: Message) => {
           "Accept": "application/vnd.github.v3+json",
         },
       }).then(response => {
-        const new_message = [
-          `Thanks <@${message.author.id}>! I've created a Gist on GitHub for you.`,
-          `(If you would prefer no gist, simply include \`!nogist\` somewhere in a message with a code block. Contact Sketch to have a gist removed if it has already been created).\n`,
-          `Code is available in this gist:`,
-          `<${response.data.html_url}>`,
-        ];
+        const new_message = `Thanks <@${message.author.id}>! I've created a Gist on GitHub for you.\n
```
Comment (ricardodevries) [inline src/events/discord/message.ts:89]: This used to be a `String[]` but v13 is not accepting that anymore: `Type 'string[]' is not assignable to type 'string'` (Accepted are string, MessagePayload and MessageOptions).
I have adapted it to a string but this needs to be reviewed to check if this is sufficient.

File: src/events/discord/messageReactionAdd.ts [hunk 3]
```diff
@@ -31,6 +39,6 @@ discord.on(
       return;
     }
 
-    guild.member(user.id)?.roles.add(reactionRole.role_id);
+    guild.members.cache.get(user.id)?.roles.add(reactionRole.role_id);
```
Comment (SketchNI) [inline src/events/discord/messageReactionAdd.ts:42]: How is this cache built? Is it provided by the discord api or built locally by the bot?

File: package.json (hunk 1 above)
Comment (SketchNI) [reply package.json:34]: To my knowledge, Heroku uses the latest current stable version with previous LTS versions available via config.

File: src/events/discord/messageReactionAdd.ts (hunk 3 above)
Comment (ricardodevries) [reply src/events/discord/messageReactionAdd.ts:42]: > How is this cache built? Is it provided by the discord api or built locally by the bot?

Those are so called managers that have been introduced in v12. It is not longer possible to directly use collection methods. See https://v12.discordjs.guide/additional-info/changes-in-v12.html#managers-cache.

Comment (SketchNI) [reply src/events/discord/messageReactionAdd.ts:42]: Ah, thank you for the link. I'm happy enough for this to be merged but it's whitep4nth3rs call ultimately.

**Review by SketchNI — APPROVED** [review]
Good commit. All changes obvious and explained thoroughly in comments.

**Pull request merged**